### PR TITLE
Allow compression of the remembered urls via zlib.

### PR DESCRIPTION
**Pull request by YPCrumble**
Uses the FANCY_COMPRESS_REMEMBERED_URLS setting.

### CODE REVIEW

**Comment by peterbe** [comment]
> @peterbe let me know your thoughts on this PR? I'd love to test this out in production soon as my cache is getting pretty large. Ideally we could include this and the recent PR's merged into a new release. Thank you and hope you're doing well!

Thank you for the ping! I needed it because it had fallen off my to-do list. 
Will review.

File: fancy_cache/memory.py [hunk 1]
```diff
@@ -12,6 +15,10 @@
 
 LOGGER = logging.getLogger(__name__)
 
+COMPRESS_REMEMBERED_URLS = getattr(
+    settings, "FANCY_COMPRESS_REMEMBERED_URLS", False
+)
```
Comment (peterbe) [inline fancy_cache/memory.py:20]: Nice! So it's entirely optional by default. That makes it feel safer to land this.

**Review by peterbe — APPROVED** [review]

**Comment by peterbe** [comment]
Sorry for being a pain, but would you mind making a (hopefully) quick that adds a bit more modern versions of Python to the `tox.ini`. 
And if you feel like it, please remove 3.7. Given that it's "end-of-life": https://devguide.python.org/versions/

So perhaps remove 3.7 from `tox.init` and https://github.com/peterbe/django-fancy-cache/blob/c0f0f1d7872cd4e1d3dd79fdf6399352440f400b/.github/workflows/ci.yml#L17 and add `"3.11"`  in both. 

Ping me if I snooze.

**Comment by YPCrumble** [comment]
Not a pain at all! Will do that now :D.

**Pull request merged**